src/components/shared/Entity/DetailsForm/useDataPlaneField.ts

```diff
@@ -75,27 +80,36 @@ export default function useDataPlaneField(
         };
     }, [intl, options]);
 
-    const evaluateDataPlane = useCallback(
-        (details: Details, selectedDataPlaneId: string | undefined) => {
-            if (selectedDataPlaneId !== storedDataPlaneId) {
-                const selectedOption = options.find(
-                    (option) => option.id === (selectedDataPlaneId ?? '')
-                );
-
-                setDetails_dataPlane(selectedOption);
+    const dataPlaneUISchema = useMemo(
+        () => ({
+            label: intl.formatMessage({
+                id: 'workflows.dataPlane.label',
+            }),
+            scope: `#/properties/${DATA_PLANE_SCOPE}`,
+            type: 'Control',
+            options: {
+                readOnly: objectRoles.length > 1 && entityName.length === 0,
+            },
```

Comment:
This is okay for now - but if this ends up living awhile we'd want to make use of the [rules api](https://jsonforms.io/docs/uischema/rules/)